node/cn/tracers/api.go

```diff
@@ -567,6 +579,13 @@ func (api *API) StandardTraceBadBlockToFile(ctx context.Context, hash common.Has
 // executes all the transactions contained within. The return value will be one item
 // per transaction, dependent on the requestd tracer.
 func (api *API) traceBlock(ctx context.Context, block *types.Block, config *TraceConfig) ([]*txTraceResult, error) {
+	if !api.unsafeTrace {
+		if atomic.LoadInt32(&heavyAPIRequestCount) >= HeavyAPIRequestLimit {
```

Comment:
The concurrency issue can be resolved like this:
```go
defer atomic.AddInt32(&heavyAPIRequestCount, -1)
if atomic.AddInt32(&heavyAPIRequestCount, 1) > HeavyAPIRequestLimit {
	return nil, fmt.Errorf("heavy debug api requests exceed the limit: %d", int64(HeavyAPIRequestLimit))
}
```

Reply:
First, concurrency issue is intended. Internally, we have concluded that it is better to go over the concurrency limit, rather to be precise to the limit. So I set the access order as `cnt read` -> `cnt write`.
As far as I understand, your suggestion is aimed at not exceeding the limit, not exactly sticking to the limit, right? Then, the original code is better fit to the intention.